apm-agent-plugins/apm-ratpack-plugin/src/main/java/co/elastic/apm/agent/ratpack/AbstractExecStarterInstrumentation.java

```diff
@@ -0,0 +1,288 @@
+                        public void accept(Transaction transaction) {
+                            transaction.activate();
+                        }
+                    });
+                }
+            });
+        }
+    }
+
+    public static class RatpackEndTransactionInstrumentation extends AbstractExecStarterInstrumentation {
+
+        public RatpackEndTransactionInstrumentation() {
+            super(named("onComplete")
+                .and(isPublic())
+                .and(takesArgument(0, Action.class))
+                .and(returns(ExecStarter.class)));
+        }
+
+        @SuppressWarnings("unused")
+        @Advice.OnMethodEnter(suppress = Throwable.class)
+        public static void onBeforeComplete(
+            @Advice.Argument(value = 0, readOnly = false) Action<? super Execution> action) {
+
+            if (tracer == null) {
+                return;
+            }
+
+            //noinspection UnusedAssignment
+            action = endTransaction(action, tracer.getConfig(WebConfiguration.class));
+        }
+
+        @VisibleForAdvice
+        public static Action<? super Execution> endTransaction(
+            final Action<? super Execution> action,
+            final WebConfiguration webConfiguration) {
+
+            return action
+                .append(new Action<Execution>() {
+                    @Override
+                    public void execute(Execution execution) throws Exception {
+
+                        execution.maybeGet(Transaction.class).ifPresent(new Consumer<Transaction>() {
+                            @Override
+                            public void accept(Transaction transaction) {
+
+                                execution.maybeGet(Context.TYPE)
+                                    .ifPresent(new FillTransactionContext(transaction, webConfiguration));
+
+                                transaction.end();
+                            }
+                        });
+                    }
+                });
+        }
+
+        private static class FillTransactionContext implements Consumer<Context> {
+
+            private final Transaction transaction;
+            private final WebConfiguration webConfiguration;
+
+            public FillTransactionContext(Transaction transaction, WebConfiguration webConfiguration) {
+                this.transaction = transaction;
+                this.webConfiguration = webConfiguration;
+            }
+
+            @Override
+            public void accept(Context context) {
+
+                transaction.withResult(context.getResponse().getStatus().toString());
+
+                fillTransactionName(transaction, context);
+                fillRequestContext(transaction.getContext().getRequest(), context);
+                fillResponseContext(transaction.getContext().getResponse(), context);
+            }
+
+            private void fillTransactionName(Transaction transaction, Context context) {
+
+                transaction.getName()
```

Comment:
The transaction name should not be the path which may contain path parameters. Rather use the url template like `GET /foo/:bar`. This makes for a better group name.